src/components/story-element.js

```diff
@@ -94,12 +94,65 @@ const DEFAULT_TEMPLATES = {
 };
 
 class StoryElementBase extends React.Component {
+
+  constructor(props){
+    super(props);
+    this.observer = null;
+    this.storyElementRef = null;
+  }
+
+  componentDidMount() {
+    this.initiateObserver();
+  }
+
+  componentWillUnmount() {
+    this.destroyObserver();
+  }
+
+  initiateObserver = () => {
+    if(this.props.disableAnalytics === true || (global && !global.qlitics)) return false;
+
+    const options = {
+      root: null,
+      rootMargin: '0px',
+      threshold: 1.0
+    };
+    this.observer = new IntersectionObserver(this.observerCallback, options);
```

Comment:
does having an individual IntersectionObserver have a perf impact? I have no idea

Reply:
Taking a call post discussion to keep it for now

* TODO : refactor later